### ISSUE TITLE
UDP Support

### ISSUE THREAD

**Issue by 200Tigersbloxed** [200Tigersbloxed/velocit#1]
Currently, only TCP is supported; however, TCP can only get us so far, especially since this is running on a Velocity backend, and not a Paper server (less use-cases for TCP). The addition of UDP tunnels would add support for some of the following plugins:

+ [Geyser](https://github.com/GeyserMC/Geyser)
+ [Plasmo-Voice](https://github.com/plasmoapp/plasmo-voice)
+ [Raknetify](https://github.com/RelativityMC/raknetify)

*etc...*

Currently, this is [waiting on an implementation](https://github.com/playit-cloud/playit-agent-java/issues/1) for the [playit-agent-java](https://github.com/playit-cloud/playit-agent-java) library.

**Comment by Clyde6790pGIT** [comment]
Are there any plans to continue this project? I cant even get port forwarding working with velocity or to even get it public.

**Comment by 200Tigersbloxed** [comment]
> Are there any plans to continue this project? I cant even get port forwarding working with velocity or to even get it public.

UDP support is still waiting for upstream changes. TCP ports should work just fine, at least the Velocity port should. If you're having an issue unrelated to UDP support, then please open a new issue, describing *in detail* the issue.

**Comment by Clyde6790pGIT** [comment]
Well, the udp implementation has had no progress for 2 years now, I practically give up on getting velocity to work on a self hosted server

**Comment by 200Tigersbloxed** [comment]
> Well, the udp implementation has had no progress for 2 years now, I practically give up on getting velocity to work on a self hosted server

If UDP is needed for your setup, then just use the [playit-agent](https://github.com/playit-cloud/playit-agent). You don't *need* a plugin implementing playit support for it to work with minecraft, you just need to proxy the ports and their protocol, which the playit agent does.

**Comment by Clyde6790pGIT** [comment]
Thanks for the advice, gonna try now.

**Comment by Clyde6790pGIT** [comment]
Just wondering, does the playit agent support velocity? I don't wanna waste my time before setting it up.
